illumina_pipeline.pl

```diff
@@ -270,6 +270,7 @@ sub checkConfig {
     elsif(! -e $opt{GENOME}){ print"ERROR: $opt{GENOME} does Not exist\n"}
     if(! $opt{SAMBAMBA_PATH}){ print "ERROR: No SAMBAMBA_PATH option found in config files.\n"; $checkFailed = 1; }
     if(! $opt{QUEUE_PATH}){ print "ERROR: No QUEUE_PATH option found in config files.\n"; $checkFailed = 1; }
+    if(! $opt{SET_HAS_METADATA}){ print "ERROR: No SET_HAS_METADATA option found in config files.\n"; $checkFailed = 1; }
```

Comment:
Are you sure you want to split them? Configuration and code often go together (just like here: the config must change when the code changes), and you still want your configuration to be versioned (and potentially tested in future).

It might indeed make sense to split them, but then you want a version vector or something. In this situation it's not clear to me why they aren't variables in the existing configuration files (other than that you don't use them, which is already a soft-fork of the pipeline logic). So the missing context for me is "why another configuration method and format?"